### PR TITLE
Fix delete enrich policy bug

### PR DESCRIPTION
**Pull request by gaobinlong**
This PR relates to #51228.
The changes are:

1. List all enrich index for the policy which will be deleted, and delete the concrete indices instead of a wildcard suffix.
2. Add `action.destructive_requires_name` cluster setting in TransportDeleteEnrichPolicyActionTests#testDeleteIsNotLocked method, ensure the test result is always correct whether the setting is set to true or not.

### CODE REVIEW

**Comment by martijnvg** [comment]
Thanks @gaobinlong for opening this PR! We will be taking a look later.

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java [hunk 1]
```diff
@@ -128,7 +129,12 @@ protected void masterOperation(
             return;
         }
 
-        deleteIndicesAndPolicy(request.getName(), ActionListener.wrap((response) -> {
+        GetIndexRequest indices = new GetIndexRequest().indices(EnrichPolicy.getBaseName(request.getName()) + "-*")
```
Comment (martijnvg) [inline x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java]: Maybe add a comment here why the delete enrich index api can't rely on just deleting all the enrich indices by a wildcard index expression?

**Review by martijnvg — APPROVED** [review]
Left a minor comment.

**Comment by martijnvg** [comment]
@elasticmachine test this please

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java (hunk 1 above)
Comment (gaobinlong) [reply x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java]: OK, I have done that.

Comment (martijnvg) [reply x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java]: thank you!

**Comment by martijnvg** [comment]
@elasticmachine test this please

**Comment by martijnvg** [comment]
Hey @gaobinlong the code style check failed. Can you run `gradlew spotlessApply`and update this pr?

**Comment by gaobinlong** [comment]
@martijnvg, OK, I have done that.

**Comment by martijnvg** [comment]
@elasticmachine test this please

**Pull request merged**